### PR TITLE
chore: added nextauth in authentication backend

### LINKED ISSUE #595: [Add] NextAuth.js should be added to Authentication
### Site URL

https://next-auth.js.org/getting-started/introduction

### Description of the above provided site

NextAuth.js is a complete open-source authentication solution for [Next.js](http://nextjs.org/) applications.
It is designed from the ground up to support Next.js and Serverless. 
*It has also been updated to natively support the latest NextJS 13 **app router** and **route handlers.***

### Flexible and easy to use

- Designed to work with any [OAuth service, it supports OAuth 1.0, 1.0A, 2.0 and OpenID Connect](https://next-auth.js.org/providers)
- Built-in support for [many popular sign-in services](https://next-auth.js.org/configuration/providers/oauth)
- Supports [email / passwordless authentication](https://next-auth.js.org/providers/email)
- Supports stateless authentication with [any backend](https://authjs.dev/reference/adapters) (Active Directory, LDAP, etc)
- Supports both JSON Web Tokens and database sessions
- Designed for Serverless but runs anywhere (AWS Lambda, Docker, Heroku, etc…)

### Own your own data

NextAuth.js can be used with or without a database.

- An open-source solution that allows you to keep control of your data
- Supports Bring Your Own Database (BYOD) and can be used with any database
- Built-in support for [MySQL, MariaDB, Postgres, SQL Server, MongoDB and SQLite](https://next-auth.js.org/configuration/databases)
- Works great with databases from popular hosting providers
- Can also be used without a database (e.g. OAuth + JWT)

*Note:* Email sign-in requires a database to be configured to store single-use verification tokens.

### Secure by default

- Promotes the use of passwordless sign-in mechanisms
- Designed to be secure by default and encourage best practices for safeguarding user data
- Uses Cross-Site Request Forgery Tokens on POST routes (sign in, sign out)
- Default cookie policy aims for the most restrictive policy appropriate for each cookie
- When JSON Web Tokens are enabled, they are encrypted by default (JWE) with A256GCM
- Auto-generates symmetric signing and encryption keys for developer convenience
- Features tab/window syncing and keepalive messages to support short-lived sessions
- Attempts to implement the latest guidance published by [Open Web Application Security Project](https://owasp.org/)

Advanced options allow you to define your own routines to handle controlling what accounts are allowed to sign in, for encoding and decoding JSON Web Tokens and to set custom cookie security policies and session properties, so you can control who can sign in and how often sessions have to be re-validated.

### Category

Backend

### Sub Category

authentication

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)
- [X] I am a GSSoC'23 contributor

### PR DESCRIPTION
**Pull request by bose-aritra2003**
## Fixes Issue

Closes #595 

## Changes proposed

Added [NextAuth.js](https://next-auth.js.org) in ***subcategory:*** **Authentication**, ***category:*** **Backend**

## Screenshots
Screenshot 2023-05-25 at 14 37 56

## Note to reviewers

### CODE REVIEW

File: database/backend/authentication.json [hunk 1]
```diff
@@ -5,5 +5,12 @@
         "url": "https://firebase.google.com/docs/auth/",
         "category": "backend",
         "subcategory": "authentication"
+    },
+    {
+        "name": "NextAuth.js",
+        "description": "NextAuth.js is a complete open-source authentication solution for Next.js applications. It is designed from the ground up to support Next.js and Serverless, and it also supports the latest Next.js 13 app router and route handlers.",
```
Comment (CBID2) [inline database/backend/authentication.json:11]: [Proofreading]
```suggestion
        "description": "NextAuth.js is a complete open-source authentication solution for Next.js applications. It is designed from the ground up to support Next.js and Serverless. It also supports the latest Next.js 13 app router and route handlers.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @bose-aritra2003! :) I pointed out a proofreading error in my inline comment. In the future, consider running your descriptions through Grammarly or Spellcheck before making a pull request. Other than that, you're off to a great start! :)

**Review by CBID2 — APPROVED** [review]
Great job @bose-aritra2003! 😄 LGTM! 😄

**Review by rupali-codes — APPROVED** [review]
looks good

**Comment by CBID2** [comment]
@rupali-codes, looks like you're going to have do a force push

**Comment by CBID2** [comment]
@rupali-codes, we need a force push

**Pull request merged**